### PR TITLE
fix: org-redirect-url-admin-update

### LINKED ISSUE #14246: [CAL-3320] fix: when updating username by admin portal - update profile too
<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-3320](https://linear.app/calcom/issue/CAL-3320/fix-when-updating-usename-by-admin-portal-update-profile-too)</sub>

### PR DESCRIPTION
**Pull request by sean-brydon**
### TL;DR

Fixes: https://github.com/calcom/cal.com/issues/14246

Related issue for updating team slugs: https://github.com/calcom/cal.com/issues/14494

<div class='graphite__hidden'>
          <div>🎥 Video uploaded on Graphite:</div>
            <a href="https://app.graphite.dev/media/video/8j8kmxIa2LzQR4eN7JKh/6c03dba5-89d8-4130-b92d-9e4b067a1dc3.mp4">
              
            </a>
          </div>
<video src="https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/8j8kmxIa2LzQR4eN7JKh/6c03dba5-89d8-4130-b92d-9e4b067a1dc3.mp4">CleanShot 2024-04-10 at 13.32.22.mp4</video>

This pull request includes changes to ensure temporary organization redirects are updated when a user's profile is updated.

### How to test?

To test, update a user's profile, ideally a username change, and check if the temporary organization redirect URLs have been updated accordingly.

Fixes CAL-3381
---

### CODE REVIEW

**Comment by sean-brydon** [comment]
* **#14493** <a href="https://app.graphite.dev/github/pr/calcom/cal.com/14493?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @sean-brydon and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by zomars — APPROVED** [review]
